pages/index.js

```diff
@@ -65,11 +79,51 @@ export default function Home({ tasks }) {
     <LayoutSytle>
       <Welcome />
       <DateStyle>{getCurrentDate()}</DateStyle>
-      <TaskCompleted tasks={tasks} handleDelete={handleDelete} />
-      <TaskOngoing tasks={tasks} handleDelete={handleDelete} />
+      {!session && (
+        <LoginWrap>
+          <LoginText>You are not signed in</LoginText>
+          <LoginButton onClick={() => signIn()}>Sign in</LoginButton>
+        </LoginWrap>
+      )}
+      {session && (
+        <>
+          <TaskCompleted tasks={tasks} handleDelete={handleDelete} />
+          <TaskOngoing tasks={tasks} handleDelete={handleDelete} />
+        </>
+      )}
```

Comment:
Normally you want to redirect the user to a login page instead of rendering different pages here. Find a solution where you use router.push("/login") or similar.